src/utils/RlGlue/rl_glue.py

```diff
@@ -30,15 +30,19 @@ def __init__(
         self,
         agent: BaseAsyncAgent,
         env: BaseAsyncEnvironment,
-        dataset_path: Path,
+        dataset_path: Path | None,
         images_save_keys: set[str] | None,
     ):
         self.environment = env
         self.agent = agent
 
-        self.dataset_path = dataset_path
-        self.is_mock_env = self.environment.__class__.__name__.startswith("Mock")
-        if not self.is_mock_env:
+        self.dataset_path = (
+            dataset_path if dataset_path is not None else Path("/data/temp")
+        )
+        self.is_mock_env = self.environment.__class__.__name__.startswith(
+            "Mock"
+        ) or self.environment.__class__.__name__.endswith("Simulator")
+        if not self.is_mock_env and dataset_path is not None:
```

Comment:
` dataset_path is not None:` will always evaluate true, dataset_path either exists or it is Path("/data/temp")